pulsar-spark/src/main/java/com/yahoo/pulsar/spark/SparkStreamingPulsarReceiver.java

```
@@ -0,0 +1,100 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.yahoo.pulsar.spark;
+
+import org.apache.spark.storage.StorageLevel;
+import org.apache.spark.streaming.receiver.Receiver;
+
+import com.yahoo.pulsar.client.api.ClientConfiguration;
+import com.yahoo.pulsar.client.api.Consumer;
+import com.yahoo.pulsar.client.api.ConsumerConfiguration;
+import com.yahoo.pulsar.client.api.Message;
+import com.yahoo.pulsar.client.api.MessageListener;
+import com.yahoo.pulsar.client.api.PulsarClient;
+import com.yahoo.pulsar.client.api.PulsarClientException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SparkStreamingPulsarReceiver extends Receiver<String> {
+
+    private ConsumerConfiguration consumerConfiguration;
+    private PulsarClient pulsarClient;
+    private String topic;
+    private String subscription;
+
+    public SparkStreamingPulsarReceiver(
+            ClientConfiguration clientConfiguration,
+            ConsumerConfiguration consumerConfiguration,
+            String url,
+            String topic,
+            String subscription
+    ) {
+        this(
+                StorageLevel.MEMORY_AND_DISK_2(),
+                clientConfiguration,
+                consumerConfiguration,
+                url,
+                topic,
+                subscription
+        );
+    }
+
+    public SparkStreamingPulsarReceiver(
+            StorageLevel storageLevel,
+            ClientConfiguration clientConfiguration,
+            ConsumerConfiguration consumerConfiguration,
+            String url,
+            String topic,
+            String subscription
+    ) {
+        super(storageLevel);
+        this.topic = topic;
+        this.subscription = subscription;
+        consumerConfiguration.setMessageListener((consumer, msg) -> {
+            store(new String(msg.getData()));
```

I think there are 2 options in case the `store()` calls fails with exception: 
 1. Don't ack the message and rely on ack-timeout to replay the message some time later (eg: 1min)
2. Just ack the message anyway and move on, basically ignoring the exception.

I prefer option 1 by default.